murdock/tests/test_job.py

```diff
@@ -268,6 +270,13 @@ def test_queued_model():
     assert job.model() == expected_model
 
 
+@pytest.mark.asyncio
+async def test_extend_job_output():
+    job = MurdockJob(commit, pr=prinfo)
+    job.notify = None
+    await job.extend_job_output("testline")
```

Comment:
I know it's obvious but better do something than nothing I think

Reply:
> After looking at the code, maybe set job.notify as MagicMock and check that it was called. Also the job.output content could be checked.

Is there a MagicMock that tests as `None`? Then we can assert that it hasn't been called. In the current case we're testing that it doesn't crash with `job.notify = None`

Reply:
I don't know, but maybe if return_value is set to None that could work ? That's worth a try :)

Reply:
That would create a function (not `None`) that returns `None`, not something that tests as `None`:

```Python
Python 3.10.8 (main, Nov  1 2022, 14:48:57) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from unittest import mock
>>> a = mock.MagicMock(return_value=None)
>>> a is None
False
>>> print(a())
None
```

Reply:
Thanks for having a look. So that's not a valid idea...

Looking that the default value of notify, it's a lambda that does nothing by default so it's never None. So maybe this `if self.notify is not None` is pointless and could just be removed ? (I don't remember why I wrote it like that... certainly a mistake)

Reply:
Also fine with me. So I can remove this test case from the PR then?

Reply:
Having something to test the notify feature would be nice if not done already. And to remove the if in the function. Maybe that's another PR.

Reply:
OK